File: 1615-maximal-network-rank/1615-maximal-network-rank.py
class Solution:
    def maximalNetworkRank(self, n: int, roads: List[List[int]]) -> int:
        connect = [[0 for _ in range(n)] for _ in range(n)]
        degree = [0 for _ in range(n)]
        
        for u, v in roads:
            degree[u] += 1
            degree[v] += 1
            connect[u][v] = 1
            connect[v][u] = 1
            
        answer = 0            
        for i in range(n):
            for j in range(n):
                if i == j:
                    continue
                if connect[i][j] == 1:
                    answer = max(answer, degree[i] + degree[j] - 1)
                else:
                    answer = max(answer, degree[i] + degree[j])
        return answer
        

File: 1615-maximal-network-rank/1615-maximal-network-rank.rs
use std::cmp;

impl Solution {
    pub fn maximal_network_rank(n: i32, roads: Vec<Vec<i32>>) -> i32 {
        let n = n as usize;
        let mut connect: Vec<Vec<usize>> = vec![vec![0; n]; n];
        let mut degree: Vec<usize> = vec![0; n];
        let mut answer = 0;
        
        for path in roads.iter() {
            let u = path[0] as usize;
            let v = path[1] as usize;
            
            connect[u][v] = 1;
            connect[v][u] = 1;
            degree[u] += 1;
            degree[v] += 1;
        }
        
        for i in 0..n {
            for j in 0..n {
                if i == j {
                    continue;
                }
                if connect[i][j] == 1 {
                    answer = cmp::max(answer, degree[i] + degree[j] - 1);
                } else {
                    answer = cmp::max(answer, degree[i] + degree[j]);
                }
            }
        }
        answer as i32
    }
}